activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -1105,21 +1119,44 @@ def reset_scope # :nodoc:
 
       private
         def find_nth_with_limit(index, limit)
```

Comment:
There's a bit of redundancy with between this definition and that in [relation/finder_methods.rb](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/relation/finder_methods.rb#L525).

This could be avoided by passing an `unsaved_records` parameter to `find_nth_with_limit`, e.g.

```ruby
# in associations/collection_proxy.rb
def find_nth_with_limit(index, limit, unsaved_records = self.unsaved_records)
  load_target if strict_loading?
  super
end
```

```ruby
# in relation/finder_methods.rb
def find_nth_with_limit(index, limit, unsaved_records = [])
  # pad non-loaded collections with unsaved records as needed here
  # ...
end
```

I avoided this path because I felt it cleaner/more elegant to avoid editing `ActiveRecord::FinderMethods#find_nth_with_limit`, but I'm ambivalent here, and open to input.